src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -3036,15 +3038,49 @@ namespace
     }
 }
 
+bool is_packed_project(const string& project_filepath, string& project_name) {
+    try 
+    {
+        vector<string> appleseed_files = get_filenames_with_extension_from_zip(project_filepath, ".appleseed");
+        
+        if (appleseed_files.size() != 1)
+            throw ExceptionUnsupportedFileFormat("Should be exactly one .appleseed file in project");
```

Comment:
This is a bit weird. You're throwing an exception but immediately catching it below.

Since this is in the renderer, and not generic code:

Instead of using exceptions, I suggest just writing an error message to the log and returning false.